### PR TITLE
add a poller for tests

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+     * @param finderFunction A Supplier functional interface for supplying the polled resource.
+     *                       Must return a java Optional.
+     * @param <T> The Type of the supplied resource
+     * @return an Optional containing the found resource, or empty if not found.
+     */
+    public <T> Optional<T> pollUntilFound(final Supplier<Optional<T>> finderFunction)  {
+
+        for (int i = 0; i < RETRY_COUNT; i++) {
+            final Optional<T> found = finderFunction.get();
+            if(found.isPresent()) {
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java]: could use ifPresent() here

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+     * @param finderFunction A Supplier functional interface for supplying the polled resource.
+     *                       Must return a java Optional.
+     * @param <T> The Type of the supplied resource.
+     * @throws AssertionError if the resource is still found after five attempts.
+     */
+    public <T> void pollUntilNotFound(final Supplier<Optional<T>> finderFunction) {
+
+        for (int i = 0; i < RETRY_COUNT; i++) {
+            final Optional<T> found = finderFunction.get();
+            if(found.isPresent()) {
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java]: ifPresent() could be used here too

File: test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/messaging/PollerTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        assertThat(found.get(), is(result));
+
+        verify(sleeper, times(4)).sleepFor(DELAY_INTERVAL_MILLIS);
+    }
+
+    @Test
+    public void shouldReturnEmptyIfNotFoundAfterFiveAttempts() throws Exception {
+
+        final Supplier<Optional<String>> supplier = mock(Supplier.class);
+
```
Comment (skiddykong) [inline test-utils/test-utils-core/src/test/java/uk/gov/justice/services/test/utils/core/messaging/PollerTest.java]: line spacing between every line of code is not necessary

File: test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+     */
+    public <T> void pollUntilNotFound(final Supplier<Optional<T>> finderFunction) {
+
+        for (int i = 0; i < retryCount; i++) {
+            final Optional<T> found = finderFunction.get();
+            if(found.isPresent()) {
+                sleeper.sleepFor(delayIntervalMillis);
+            }
+        }
+
```
Comment (amitxsethi) [inline test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/messaging/Poller.java]: We should break from this loop if "found" is empty, to remove unnecessary calls to finderFunction.get()

**Pull request merged**